### DISCUSSION TITLE
Audio Speed is too fast with snapcast - but alsasink is working

### DISCUSSION THREAD

**Discussion by johnnyeightpence** [snapcast/snapcast#980] [Q&A]
I am running mopidy on a Raspberry PI 4 and I use snapcast for a multiroom audio. 
When I configure mopidy to use alsasink device=hw:0,0 the sound is perfect.
When I use filesink location=/tmp/snapfifo the audio file is played about 5 times too fast. 
Both samplerates are the same.

Here is my mopidy config for using Snapcast:
output = audioresample ! audioconvert ! audio/x-raw,rate=48000,channels=2,format=S16LE ! filesink location=/tmp/snapfifo 

In the Snapcast config I only changed this in [stream]: 
stream = pipe:///tmp/snapfifo?name=Mopidy
sampleformat = 48000:16:2

Any ideas on how to solve this? Thanks for any suggestion.

**Comment by kingosticks** [comment]
Do you need to solve it? There are four ways to get the audio from Mopidy to Snapserver (all in the docs on here) and filesink is the worst option.

**Reply by johnnyeightpence** [reply]
Hi
I am a super noob in all this. 
Where do I find the documentation? Is this like a wiki or a pdf? 
I just followed a random online tutorial on "multi room audio" and wanted to implement it accordingly. Now I am stuck. I wouldnt mind trying it any other way.

**Reply by Pajn** [reply]
There are a lot of useful info in the doc folder. I think the alsa loopback configuration would be the best one for you https://github.com/badaix/snapcast/blob/master/doc/configuration.md#alsa

**Reply by kingosticks** [reply]
The full docs are linked from the readme at https://github.com/badaix/snapcast/blob/master/README.md#server. There are two TCP options and an ALSA loopback option, the latter is probably your best bet as suggested by @Pajn. If you have not yet read the readme you should definitely do that, lots of useful and up to date info in there.

The general problem with random online tutorials is they never get updated by the author. Once upon a time filesink was the only option but fortunately there are now other, better options available. The other problem with those sorts of tutorials is they are sometimes totally wrong. Best bet when using any piece of software is to start with the official documentation and treat everything else with a healthy dose of suspicion!

**Comment by johnnyeightpence** [comment]
Hi you both. Thanks a lot for the fast feedback. I am digging into the documentation now and try to figure out what TCP does etc. 
It will take a couple of days until I figured all that out.
Can you recommend a good description on how the system works with TCP?
I read the instructions but I am confused with the server and client config?
For me its not so clear what the role of mopidy is in this setup. Is mopidy a server? 
Be patient with me, please. I am pretty new to all this and really want to understand and not just copy what a guy from a tutorial tells me to :)

**Reply by kingosticks** [reply]
Try the alsa loopback device then, that's conceptually simpler. If it works, problem solved.

**Reply by johnnyeightpence** [reply]
That did the trick! 
Thank you all for the support!
I learned a lot: 
- there is no perfect tutorial because updates are often neglected
- if there is good documentation its worth reading
- its more fun knowing the background and gives more flexibility in case the system changes
- sharing is caring
